### PR TITLE
Test operator precedence

### PR DESCRIPTION
**Pull request by ushirask**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/testSuite/Vintop02.bal [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import ballerina/io;
+
+public function main() {
+  printBoolean(7 == 8); // @output 0
```
Comment (ushirask) [inline compiler/testSuite/Vintop02.bal]: @jclark this fails with error `error: intersection of operands of operator '==' is empty` but `7 + 1 == 8` does not.

Comment (manuranga) [reply compiler/testSuite/Vintop02.bal]: In that case this file should have a name starting with with `F`

Comment (ushirask) [reply compiler/testSuite/Vintop02.bal]: Fixed.

Comment (ushirask) [reply compiler/testSuite/Vintop02.bal]: > In that case this file should have a name starting with with `F`

Fixed.

**Review by jclark — COMMENTED** [review]
Please don't include the .ll files

File: compiler/testSuite/Fintop00.bal [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import ballerina/io;
+
+public function main() {
+  printBoolean(7 == 8); // @output 0
```
Comment (jclark) [inline compiler/testSuite/Fintop00.bal]: This is not legal according to the Ballerina spec. https://ballerina.io/spec/lang/master/#section_6.29

> it is a compile time error if the intersection of the static types of the operands is empty

which in this case it is

**Review by jclark — CHANGES_REQUESTED** [review]
Please leave out the .ll files

**Comment by ushirask** [comment]
> Please leave out the .ll files

Removed.

**Comment by manuranga** [comment]
Did you remove the `F` case you had? It's still good to have that, let's convert that to an `E` case.

**Comment by ushirask** [comment]
> Did you remove the `F` case you had? It's still good to have that, let's convert that to an `E` case.

Added as an error test

File: compiler/testSuite/Eintop00.bal [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+
+function printBoolean(boolean x) {
+  if x {
+    io:println(1);
+  } else {
+    io:println(0);
+  }
+}
```
Comment (jclark) [inline compiler/testSuite/Eintop00.bal]: Better to end files with a newline. Not having one confuses some tools.

**Review by jclark — COMMENTED** [review]
You still needs tests for the precedence of unary operators e.g. `-1 + 2` parses as `(-1) + 2`.

`!true == false` parses as `(!true) == false`.

**Pull request merged**